### PR TITLE
Add tests for data source selector

### PR DESCRIPTION
**Pull request by mengweieric**
### Description

Add e2e tests for data source selector.

```
  Data source selector
    ✓ displays all data sources by default (5755ms)
    ✓ filters options based on user input
    ✓ updates the visual length of the dropdown based on filtered results
    ✓ selects the correct option when clicked


  4 passing (9s)
```

### Issues Resolved

Following up on [#5211](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5211)

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/data_source_selector.spec.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+const miscUtils = new MiscUtils(cy);
+const testFixtureHandler = new TestFixtureHandler(
+  cy,
+  Cypress.env('openSearchUrl')
+);
+
+describe('Data source selector', () => {
+  before(() => {
+    testFixtureHandler.importJSONMapping(
```
Comment (kavilla) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/data_source_selector.spec.js:19]: Haven't merged the `main` pr yet but could you add the setting of the global tenant?

https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/963/files. Without the security plugin this is fine but once the security plugin is added the default tenant is the private tenant and not the `.kibana`

**Comment by kavilla** [comment]
Mind adding this `apps/data_explorer/data_source_selector.spec.js`

https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/package.json#L35  to make it a little bit faster to run.

**Review by kavilla — APPROVED** [review]
ty

**Review by ashwin-pc — APPROVED** [review]

**Comment by ruanyl** [comment]
@mengweieric should this backport to 2.x?

**Comment by mengweieric** [comment]
> @mengweieric should this backport to 2.x?

yes, please add backport 2.x label thanks

**Comment by wbeckler** [comment]
For these tests, are they written in such a way that they would have detected the malfunctions reported here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5609

**Pull request merged**